src/rules/tsrDetectPossibleTimingAttacksRule.ts

```diff
@@ -1,19 +1,79 @@
 import * as ts from 'typescript';
 import * as Lint from 'tslint';
 
-const keywordMask = new RegExp('^((' + [
+const keywordMask = new RegExp('^.*((' + [
     'password',
     'secret',
     'api',
     'apiKey',
     'token',
-    'auth',
     'pass',
     'hash'
-].join(')|(') + '))$', 'im');
+].join(')|(') + ')).*$', 'im');
 
 function containsKeyword (node: ts.Expression): boolean {
-    return Boolean(node && node.kind === ts.SyntaxKind.Identifier && keywordMask.test((node as ts.Identifier).text));
+    switch (node.kind) {
+        case ts.SyntaxKind.CallExpression:
+            return containsKeywordCallExpression(node as ts.CallExpression);
+        case ts.SyntaxKind.ElementAccessExpression:
+            return containsKeywordElementAccessExpression(node as ts.ElementAccessExpression);
+        case ts.SyntaxKind.Identifier:
+            return containsKeywordIdentifier(node as ts.Identifier);
+        case ts.SyntaxKind.PropertyAccessExpression:
+            return containsKeywordPropertyAccessExpression(node as ts.PropertyAccessExpression);
+        default:
+            return false;
+    }
+}
+
+function containsKeywordCallExpression (node: ts.CallExpression) {
+    return containsKeyword(node.expression);
+}
+
+function containsKeywordElementAccessExpression (node: ts.ElementAccessExpression) {
+    if (node.argumentExpression.kind === ts.SyntaxKind.StringLiteral) {
+        const argumentExpression = (node.argumentExpression as ts.StringLiteral);
+        return containsKeyword(node.expression) || Boolean(keywordMask.test(argumentExpression.text));
+    } else {
+        return containsKeyword(node.expression) || containsKeyword(node.argumentExpression);
+    }
+}
+
+function containsKeywordIdentifier (node: ts.Identifier) {
+    return Boolean(keywordMask.test(node.text));
+}
+
+function containsKeywordPropertyAccessExpression (node: ts.PropertyAccessExpression) {
+    return containsKeyword(node.expression) || containsKeyword(node.name);
+}
+
+function isVulnerableType (node: ts.Expression): boolean {
+    switch (node.kind) {
+        case ts.SyntaxKind.CallExpression:
+            return isVulnCallExpression(node as ts.CallExpression);
```

Comment:
I think it's better to have more readable function names: `isVuln*` => `isVulnerable*`